webapp/channels/src/components/admin_console/team_channel_settings/group/index.ts

```diff
@@ -32,13 +26,12 @@ function mapStateToProps(state: GlobalState, ownProps: OwnProps) {
     };
 }
 
-function mapDispatchToProps(dispatch: Dispatch) {
-    return {
-        actions: bindActionCreators<ActionCreatorsMapObject<ActionFunc>, Actions>({
-            getData: () => getGroups(),
-        }, dispatch),
-    };
-}
+const mapDispatchToProps = {
+    getData: () => Promise.resolve(),
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
 
-export default connect(mapStateToProps, mapDispatchToProps)(List);
+export type PropsFromRedux = ConnectedProps<typeof connector>;
 
```

Comment:
@M-ZubairAhmed is this what you had in mind? I am getting the following error in `channel_groups.tsx` and `team_groups.tsx` after making the changes: `Property 'actions' is missing in type '{ team: Team; isModeSync: boolean; groups: Group[]; totalGroups: number; onGroupRemoved: (groupId: string) => void; setNewGroupRole: (groupId: string) => void; type: string; isDisabled: boolean | undefined; }' but required in type 'Omit<ClassAttributes<GroupList> & Props, "data" | "total" | "emptyListTextId" | "emptyListTextDefaultMessage" | "removeGroup" | "setNewGroupRole">'`;

basically `GroupList` is now expecting its parent components to pass an implementation of `getData: () => Promise<Group[]>` as a prop value to `GroupList` 's `actions` prop.

Reply:
Please do not run any integration and end-to-end test at this point.

Reply:
@ridwankabeer435 i have pulled this branch locally and here are my observations/suggestions

- As seen in https://github.com/mattermost/mattermost/pull/23511#discussion_r1245502729 Lets export this ownProps as well.
- In https://github.com/mattermost/mattermost/pull/23511#discussion_r1245503014 so essentially you Props will just be a combinations of OwnProps and PropsFromRedux
- You might need to add more ownProp types in index file
- Cross check the function return signature of mapDispatchToProps https://github.com/mattermost/mattermost/pull/23511#discussion_r1245504798 the new one should be the same
- Because this is a weird case lets hard type cast the `getData: () => Promise.resolve() as any,` 

I wrote everything here to condense the message clearly in one place